source/common/upstream/upstream_impl.cc

```diff
@@ -459,8 +460,12 @@ HostDescription::SharedConstAddressVector HostDescriptionImplBase::makeAddressLi
 
 Network::UpstreamTransportSocketFactory& HostDescriptionImplBase::resolveTransportSocketFactory(
     const Network::Address::InstanceConstSharedPtr& dest_address,
-    const envoy::config::core::v3::Metadata* metadata) const {
-  auto match = cluster_->transportSocketMatcher().resolve(metadata);
+    const envoy::config::core::v3::Metadata* endpoint_metadata) const {
+  metadata_mutex_.Lock();
```

Comment:
Why the new locking?

Reply:
My original reasoning was that the `locality_metadata_` member variable is read inside the function and I was treating it like the endpoint metadata, which is lock-protected. After looking at this again, it appears to be unnecessary. The endpoint metadata can change during the lifetime of the HostDescription, but the locality metadata does not.

I just made it const and removed the locking. Thanks